### PR TITLE
try go fix cargo fmt invocation

### PR DESCRIPTION
**Pull request by matthiaskrgr**
Use absolute path $CARGO_HOME/bin/cargo to get cargo path in rustfmt tests.

I get the same error as this is supposed to fix on my system because I have an alias
`cargo="cargo --foo` on my system.

This causes `cargo +nightly check` to be resolved to `cargo --foo +nightly check` which is wrong because the `+nightly` needs to be the first arg!
Work around by circumventing possible aliasing and using the absolute path to the cargo binary.

changelog: none

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
@bors try 
let's see if this works on appveyor...

**Comment by matthiaskrgr** [comment]
Hm, apparently `C:\Users\appveyor\.cargo\bin\cargo` is not the location of the cargo binary on appveyor...?

File: clippy_dev/src/fmt.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+use home;
```
Comment (lzutao) [inline clippy_dev/src/fmt.rs]: We don't need this in 2018 edition
```suggestion

```

File: clippy_dev/src/fmt.rs [hunk 2]
```diff
@@ -31,6 +32,26 @@ struct FmtContext {
...
+    let mut args = vec!["+nightly", "fmt", "--all"];
+    if context.check {
+        args.push("--");
+        args.push("--check");
+    }
+    // in case something aliases cargo=cargo --foo, the order of args will be messed up
+    // because we will end up with cargo --foo +nightly fmt but the "+nightly" needs to come first
+    // get the absolute path to the cargo binary inside $CARGO_HOME
+    let cargo_home = home::cargo_home().expect("failed to get cargo home path!");
+    let cargo_path = cargo_home.join("bin").join("cargo");
```
Comment (lzutao) [inline clippy_dev/src/fmt.rs]:
```suggestion
    let cargo_path = cargo_home.join("bin").join(format!("cargo{}", std::env::consts::EXE_SUFFIX));
```

**Comment by matthiaskrgr** [comment]
The issue turned out to be a cargo bug: https://github.com/rust-lang/cargo/issues/7475

**Pull request closed without merging**